### PR TITLE
Track deprecated arguments in @envelop/prometheus

### LINKED ISSUE #1476: @envelop/prometheus doesn't increment graphql_envelop_deprecated_field for mutation arguments
### Issue workflow progress

_Progress of the issue based on the [Contributor Workflow](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md#a-typical-contributor-workflow)_

- [ ] 1. The issue provides a [minimal reproduction](https://en.wikipedia.org/wiki/Minimal_reproducible_example) available on [Stackblitz](https://stackblitz.com/fork/node).
  - _Please install the latest `@envelop/*` packages that you are using._
  - _Please make sure the reproduction is as small as possible._
  - I provided a reproduction through a linked PR with two tests.  Let me know if it _has_ to be through Stackblitz.
- [x] 2. A failing test has been provided
- [ ] 3. A local solution has been provided
- [x] 4. A pull request is pending review

---

**Describe the bug**

`@envelop/prometheus` keep metrics for deprecated fields queried but doesn't track deprecated arguments and deprecated fields on inputs.

**To Reproduce**
Given the schema
```graphql
      type Query { }
      input MutationInput {
        deprecatedField: String @deprecated(reason: "old")
        regularField: String!
      }
      type MutationPayload {
        payloadField: String
      }
      type Mutation {
        mutationWithDeprecatedFields(
          nonDeprecatedInput: MutationInput
          deprecatedInput: String @deprecated(reason: "old")
        ): MutationPayload
      }
```

Calling the mutation with either `deprecatedInput` or with `nonDeprecatedInput.deprecatedField` will not update `graphql_envelop_deprecated_field`.

**Expected behavior**

`graphql_envelop_deprecated_field` should be updated with an entry for both deprecations.

**Environment:**

- OS:
- NodeJS:
- `@envelop/` versions:
  - `@envelop/prometheus`: `6.4.2`

### PR DESCRIPTION
**Pull request by AugustinLF**
## Description

Add support for tracking deprecated arguments in `@envelop/prometheus`. I could not add support for input fields since my assumption is that we can't catch those at parse time, which doesn't work with the given set-up.

Fixes #1476 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test `Should track deprecated arguments in mutation` has been added to `packages/plugins/prometheus/test/prom.spec.ts`
- [ ] Skipped test `Should track deprecated fields in mutation input` has been added to `packages/plugins/prometheus/test/prom.spec.ts`

**Test Environment**:

- OS:
- `@envelop/...`: !6.4.2
- NodeJS: v14.18.2

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

As explained in the description, I cover only one of the two problems. I can either try to cover both cases but would need some guidance on where to hook to see what parameters are passed to the mutation. Or I can remove the skipped test and stick with arguments for now, which is already an improvement, and work on the inputs in a follow-up PR.

### CODE REVIEW

File: packages/plugins/prometheus/test/prom.spec.ts [hunk 1]
```diff
@@ -494,6 +514,72 @@ describe('Prom Metrics plugin', () => {
       expect(await metricCount('graphql_envelop_execute_resolver', 'count')).toBe(2);
       expect(await metricCount('graphql_envelop_deprecated_field')).toBe(1);
     });
+
+    it.skip('Should track deprecated fields in mutation input', async () => {
```
Comment (n1ru4l) [inline packages/plugins/prometheus/test/prom.spec.ts]: Can you remove this test case if the implementation is not covered?

**Review by n1ru4l — CHANGES_REQUESTED** [review]
We don't need a skipped test, you can remove it.

File: packages/plugins/prometheus/test/prom.spec.ts (hunk 1 above)
Comment (n1ru4l) [reply packages/plugins/prometheus/test/prom.spec.ts]: This could be tracked with the https://www.envelop.dev/plugins/use-extended-validation plugin 🤔

Comment (AugustinLF) [reply packages/plugins/prometheus/test/prom.spec.ts]: Thanks, I'll remove the skipped test. And I'll have a look if `use-extended-validation` could work in our case 👀

**Comment by AugustinLF** [comment]
@n1ru4l I updated the tests to only cover deprecated arguments :)

**Pull request merged**